components/pluralrules/src/lib.rs

```diff
@@ -0,0 +1,381 @@
+//! assert_eq!(pr.select(5_usize), PluralCategory::Other);
+//! ```
+//!
+//! ## Plural Rules
+//!
+//! The crate provides the main struct [`PluralRules`] which handles selection
+//! of the correct [`Plural Category`] for a given language and [`Plural Type`].
+//!
+//! ## Plural Category
+//!
+//! Every number in every language belongs to a certain [`Plural Category`].
+//! For example, Polish language uses four:
+//!
+//! * [`One`](./enum.PluralCategory.html#variant.One): `1 miesiąc`
+//! * [`Few`](./enum.PluralCategory.html#variant.Few): `2 miesiące`
+//! * [`Many`](./enum.PluralCategory.html#variant.Many): `5 miesięcy`
+//! * [`Other`](./enum.PluralCategory.html#variant.Other): `1.5 miesiąca`
+//!
+//! ## Plural Rule Type
+//!
+//! Plural rules depend on the use case. This crate supports two types of plural rules:
+//!
+//! * [`Cardinal`](./enum.PluralRuleType.html#variant.Cardinal): `3 doors`, `1 month`, `10 dollars`
+//! * [`Ordinal`](./enum.PluralRuleType.html#variant.Ordinal): `1st place`, `10th day`, `11th floor`
+//!
+//! ## Data Provider
+//!
+//! In order to function, the API requires data from [`CLDR`].
+//!
+//! [`ICU4X`] is going to use a special API for handling data management called `DataProvider`.
+//! Until that happens, this crate will provide a simple `JSON` and `bincode` providers behind a
+//! flag.
+//! For tests and documentation examples, there is also a `DummyDataProvider`.
+//!
+//! All of the content of the [`data`] module is heavily experimental and subject to change.
+//!
+//! [`ICU4X`]: https://github.com/unicode-org/icu4x
+//! [`PluralRules`]: ./struct.PluralRules.html
+//! [`PluralRules`]: ./struct.PluralRules.html
+//! [`Plural Type`]: ./enum.PluralRuleType.html
+//! [`Plural Category`]: ./enum.PluralCategory.html
+//! [`Language Plural Rules`]: https://unicode.org/reports/tr35/tr35-numbers.html#Language_Plural_Rules
+//! [`CLDR`]: http://cldr.unicode.org/
+//! [`data`]: ./data/index.html
+pub mod data;
+mod operands;
+pub mod rules;
+
+use icu_locale::LanguageIdentifier;
+pub use operands::PluralOperands;
+
+/// A type of a plural rule which can be associated with the [`PluralRules`] struct.
+///
+/// [`PluralRules`]: ./struct.PluralRules.html
+#[derive(Debug, PartialEq, Eq, Clone, Copy, Hash)]
+pub enum PluralRuleType {
+    /// Ordinal plural forms denote the order of items in a set and are always integers.
+    ///
+    /// For example, English has four forms for ordinals:
+    ///
+    /// * [`One`]: `1st floor`, `21st floor`, `101st floor`
+    /// * [`Two`]: `2nd floor`, `22nd floor`, `102nd floor`
+    /// * [`Few`]: `3rd floor`, `23rd floor`, `103rd floor`
+    /// * [`Other`]: `4th floor`, `11th floor`, `96th floor`
+    ///
+    /// [`One`]: ./enum.PluralCategory.html#variant.One
+    /// [`Two`]: ./enum.PluralCategory.html#variant.Two
+    /// [`Few`]: ./enum.PluralCategory.html#variant.Few
+    /// [`Other`]: ./enum.PluralCategory.html#variant.Other
+    Cardinal,
```

Comment:
I think the source comments accidentally came along for the ride.  Just swap line 105 with line 116, I think that should do it.

Reply:
yep! thank you! fixed!